lib/ansible/modules/net_tools/basics/uri.py

```diff
@@ -317,6 +336,38 @@ def absolute_location(url, location):
         return location
 
 
+def kv_list(data):
+    ''' Convert data into a list of key-value tuples '''
+    if data is None:
+        return None
+
+    if isinstance(data, Sequence):
+        return list(data)
+
+    if isinstance(data, Mapping):
+        return list(data.items())
+
+    raise ValueError('cannot form-urlencode body, expect list or dict')
```

Comment:
Can we make this a `TypeError` to match the error that `urlencode` would itself give?